step-automation-packages/step-automation-packages-support/src/main/java/step/automation/packages/AutomationPackageArchive.java

```diff
@@ -39,25 +41,28 @@ public class AutomationPackageArchive implements Closeable {
     private final File originalFile;
     private final String originalFileName;
     private boolean internalClassLoader = false;
+    private final AutomationPackageArchiveType type;
 
     public AutomationPackageArchive(ClassLoader classLoader) {
         this.classLoader = classLoader;
         this.originalFile = null;
         this.originalFileName = null;
+        this.type = JAVA;
```

Comment:
@david-stephan 
Shouldn't we also support the `type` field in yaml descriptor for AP?